src/nxt_unit.c

```diff
@@ -3224,23 +3224,20 @@ nxt_unit_buf_read(nxt_unit_buf_t **b, uint64_t *len, void *dst, size_t size)
     while (buf != NULL) {
         last_buf = buf;
 
-        copy = buf->end - buf->free;
-        copy = nxt_min(rest, copy);
+        copy = nxt_min(rest, buf->end - buf->free);
 
         p = nxt_cpymem(p, buf->free, copy);
 
         buf->free += copy;
         rest -= copy;
 
-        if (rest == 0) {
-            if (buf->end == buf->free) {
-                buf = nxt_unit_buf_next(buf);
```

Comment:
If we need to touch the code, we should first ensure it has an issue.
And `buf` is not used after the block, right?
```
diff -r 3d1aecfc7ceb src/nxt_unit.c
--- a/src/nxt_unit.c    Wed May 24 14:01:28 2023 +0800
+++ b/src/nxt_unit.c    Wed May 24 14:33:18 2023 +0800
@@ -3233,10 +3233,6 @@
         rest -= copy;

         if (rest == 0) {
-            if (buf->end == buf->free) {
-                buf = nxt_unit_buf_next(buf);
-            }
-
             break;
         }
```
Do you know why?

Reply:
Actually, your code is equal to this:
```
        if (buf->end == buf->free) {
            buf = nxt_unit_buf_next(buf);
        }

        if (rest == 0) {
            break;
        }
```
==
```
        if (rest == 0) {
            break;
        }

         if (buf->end == buf->free) {
            buf = nxt_unit_buf_next(buf);
        }
```
==
```
        if (rest == 0) {
            break;
        }

         if (rest != 0) {
            buf = nxt_unit_buf_next(buf);
        }
```
==
```
        if (rest == 0) {
            break;
        }

         buf = nxt_unit_buf_next(buf);
```

But it's better to understand why `buf` is not used after the block before touching it.